internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -1189,12 +1207,12 @@ func (_ *Server) repositoryAdded(
 func (s *Server) fetchRepo(
 	ctx context.Context,
 	repo *repo,
-) (*db.Repository, error) {
+) (*models.EntityWithProperties, error) {
 	l := zerolog.Ctx(ctx)
 
-	dbrepo, err := s.store.GetRepositoryByRepoID(ctx, repo.GetID())
+	repoEnt, err := s.repos.RefreshRepositoryByUpstreamID(ctx, repo.GetID())
```

Comment:
perhaps not for this PR. But one TODO time we should add is figuring out what to do if we get multiple repositories out of this call and handling that. e.g. say we have several provider classes, they might have an upstream ID that clashes, we need to be able to handle that.

Reply:
Good call, we need to search by provider, too